apps/api-extractor/src/collector/Collector.ts

```diff
@@ -599,9 +600,43 @@ export class Collector {
       declarationMetadata.isSealed = modifierTagSet.isSealed();
       declarationMetadata.isVirtual = modifierTagSet.isVirtual();
 
-      // Require the summary to contain at least 10 non-spacing characters
-      declarationMetadata.needsDocumentation = !tsdoc.PlainTextEmitter.hasAnyTextContent(
-        parserContext.docComment.summarySection, 10);
+      if (modifierTagSet.hasTag(AedocDefinitions.preapprovedTag)) {
+        // This feature only makes sense for potentially big declarations.
+        switch (astDeclaration.declaration.kind) {
+          case ts.SyntaxKind.ClassDeclaration:
+          case ts.SyntaxKind.EnumDeclaration:
+          case ts.SyntaxKind.InterfaceDeclaration:
+          case ts.SyntaxKind.ModuleDeclaration:
+            if (declaredReleaseTag === ReleaseTag.Internal) {
+              declarationMetadata.isPreapproved = true;
+            } else {
+              this.messageRouter.addAnalyzerIssue(
+                ExtractorMessageId.PreapprovedBadReleaseTag,
+                `The @preapproved tag cannot be applied to "${astDeclaration.astSymbol.localName}"`
+                  + ` without an @internal release tag`,
+                astDeclaration
+              );
+            }
+            break;
+          default:
+            this.messageRouter.addAnalyzerIssue(
+              ExtractorMessageId.PreapprovedUnsupportedType,
+              `The @preapproved tag cannot be applied to "${astDeclaration.astSymbol.localName}"`
+                + ` because it is not a supported declaration type`,
+              astDeclaration
+            );
+            break;
+        }
+      }
+
+      if (astDeclaration.declaration.kind === ts.SyntaxKind.Constructor) {
+        // NOTE: If the constructor summary is missing, then ApiModelGenerator will auto-generate one.
+        declarationMetadata.needsDocumentation = false;
+      } else {
+        // Require the summary to contain at least 10 non-spacing characters
+        declarationMetadata.needsDocumentation = !tsdoc.PlainTextEmitter.hasAnyTextContent(
```

Comment:
Seems a little arbitrary

Reply:
@iclanton Seems like it needs to be somewhat arbitrary.  I'm open to ideas for tuning the number.  We could also make it configurable, but not until AE7 ships.

With 10 characters, this counts as documentation:
```ts
/**
 * Returns the width.
 */
export function getWidth() { }
```

But this does not:
```ts
/**
 * Width.
 */
export function getWidth() { }
```

That was the basic idea behind it.

Reply:
@KevinTCoughlin and @natalieethell were saying that some comments in OUIFR wouldn't meet this standard, so they'd have to disable this warning. That's a little concerning to me.

Reply:
Yea, we'll need to pay the cost at some point. It'll either be by suppressing the warning and upgrading OR handling it as part of the upgrade.

I imagine folks will request this to be configurable at least. I don't have a strong opinion on length.

Reply:
> @KevinTCoughlin and @natalieethell were saying that some comments in OUIFR wouldn't meet this standard, so they'd have to disable this warning.

@iclanton This feature is designed specifically to address that concern:  Rather than generating a regular warning that you'd have to suppress, it simply labels the API as `// (undocumented)` in the .api.md file.  It's pretty noninvasive.  You can start out with hundreds of these labels.  Over time as you develop the library, you can fill in the docs at your own pace until all the labels are finally eliminated.

At least that was the intent. Let us know if you think the design can be improved.